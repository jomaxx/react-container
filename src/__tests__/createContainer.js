import React from "react";
import { render } from "react-testing-library";
import { createContainer } from "../";

it("has initial state", () => {
  const Container = createContainer({ done: true });
  const spy = jest.fn(() => null);

  render(<Container>{spy}</Container>);

  expect(spy).toHaveBeenCalledTimes(1);
  expect(spy).toMatchSnapshot();
});

it("derives state from props", () => {
  const Container = createContainer();
  const spy = jest.fn(() => null);

  Container.getDerivedStateFromProps = () => ({
    done: true
  });

  render(<Container>{spy}</Container>);

  expect(spy).toHaveBeenCalledTimes(1);
  expect(spy).toMatchSnapshot();
});
