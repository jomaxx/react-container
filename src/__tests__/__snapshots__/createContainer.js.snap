// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`derives state from props 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "setState": [Function],
        "state": Object {
          "done": true,
        },
      },
    ],
  ],
}
`;

exports[`has initial state 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "setState": [Function],
        "state": Object {
          "done": true,
        },
      },
    ],
  ],
}
`;
